components/tidb_query_vec_executors/src/runner.rs

```diff
@@ -370,6 +370,10 @@ impl<SS: 'static> BatchExecutorsRunner<SS> {
 
         let mut time_slice_start = Instant::now();
         loop {
+            // TODO: Use more precise method to detect the deadline.
+            // Check if we reach the dealine
+            self.deadline.check()?;
```

Comment:
Looks like there is not much code from the loop end to loop start, do we still need to check it?